changelogs/fragments/6756.yml

```
@@ -0,0 +1,2 @@
+fix:
+- Show a error toast when workspace read only user delete saved objects ([#6756](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/6756))
```

```suggestion
- Show error toast when fail to delete saved objects ([#6756](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/6756))
```
The error toast should be a generic solution when fail to delete saved objects, no need to restrict to workspace permission failure.